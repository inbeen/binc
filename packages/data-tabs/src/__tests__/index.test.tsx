import { describe, expect, it, vi } from 'vitest';
import { fireEvent, render } from '@testing-library/react';
import { DataTabs } from '..';
import { flatItems, nestItems } from './data';

// Mock ResizeObserver
global.ResizeObserver = vi.fn().mockImplementation(() => ({
  observe: vi.fn(),
  unobserve: vi.fn(),
  disconnect: vi.fn()
}));

describe('DataTabs', () => {
  it('should flat items work', () => {
    const { asFragment, container } = render(<DataTabs activeKey="c" items={flatItems} />);
    expect(asFragment().firstChild).toMatchSnapshot();
    expect(container.querySelectorAll('.binc-data-tabs__group').length).toBe(5);
  });

  it('shoule nest items work', () => {
    const { asFragment, container } = render(<DataTabs activeKey="c" items={nestItems} />);
    expect(asFragment().firstChild).toMatchSnapshot();
    expect(container.querySelectorAll('.binc-data-tabs__group').length).toBe(3);
  });

  it('should trigger event correct', () => {
    let changeKey = '';
    const handleChange = (key: string) => (changeKey = key);
    const { container } = render(<DataTabs items={flatItems} onChange={handleChange} />);
    fireEvent.click(container.querySelector('.binc-data-tabs__item')!);
    expect(changeKey).toBe('total');
  });

  it('should no trigger event when click active one', () => {
    let changeKey = '';
    const handleChange = (key: string) => (changeKey = key);
    const { container } = render(<DataTabs activeKey="total" items={flatItems} onChange={handleChange} />);
    fireEvent.click(container.querySelector('.binc-data-tabs__item')!);
    expect(changeKey).toBe('');
  });
});
