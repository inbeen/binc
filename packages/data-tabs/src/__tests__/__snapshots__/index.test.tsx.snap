// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DataTabs > should flat items work 1`] = `
<div
  class="binc-data-tabs"
>
  <div
    class="binc-data-tabs__group"
  >
    <div
      class="binc-data-tabs__item"
      style="width: 120px;"
    >
      <div
        class="binc-data-tabs__item__header"
      >
        <div
          class="binc-data-tabs__item__title"
        >
          总览
        </div>
      </div>
      <div
        class="binc-data-tabs__item__value"
      >
        123456789
      </div>
    </div>
  </div>
  <div
    class="binc-data-tabs__group"
  >
    <div
      class="binc-data-tabs__item"
      style="width: 120px;"
    >
      <div
        class="binc-data-tabs__item__header"
      >
        <div
          class="binc-data-tabs__item__title"
        >
          条件A
        </div>
      </div>
      <div
        class="binc-data-tabs__item__value"
      >
        100000000
      </div>
    </div>
  </div>
  <div
    class="binc-data-tabs__group"
  >
    <div
      class="binc-data-tabs__item"
      style="width: 120px;"
    >
      <div
        class="binc-data-tabs__item__header"
      >
        <div
          class="binc-data-tabs__item__title"
        >
          条件B
        </div>
      </div>
      <div
        class="binc-data-tabs__item__value"
      >
        23000000
      </div>
    </div>
  </div>
  <div
    class="binc-data-tabs__group"
  >
    <div
      class="binc-data-tabs__item binc-data-tabs__item--active"
      style="width: 120px;"
    >
      <div
        class="binc-data-tabs__item__header"
      >
        <div
          class="binc-data-tabs__item__title"
        >
          条件C
        </div>
      </div>
      <div
        class="binc-data-tabs__item__value"
      >
        456000
      </div>
    </div>
  </div>
  <div
    class="binc-data-tabs__group"
  >
    <div
      class="binc-data-tabs__item"
      style="width: 120px;"
    >
      <div
        class="binc-data-tabs__item__header"
      >
        <div
          class="binc-data-tabs__item__title"
        >
          非常长的条件，测试溢出情况
        </div>
      </div>
      <div
        class="binc-data-tabs__item__value"
      >
        789
      </div>
    </div>
  </div>
</div>
`;

exports[`DataTabs > shoule nest items work 1`] = `
<div
  class="binc-data-tabs"
>
  <div
    class="binc-data-tabs__group"
  >
    <div
      class="binc-data-tabs__item"
      style="width: 120px;"
    >
      <div
        class="binc-data-tabs__item__header"
      >
        <div
          class="binc-data-tabs__item__title"
        >
          总览
        </div>
      </div>
      <div
        class="binc-data-tabs__item__value"
      >
        123456789
      </div>
    </div>
  </div>
  <div
    class="binc-data-tabs__group"
  >
    <div
      class="binc-data-tabs__item"
      style="width: 120px;"
    >
      <div
        class="binc-data-tabs__item__header"
      >
        <div
          class="binc-data-tabs__item__title"
        >
          条件A
        </div>
      </div>
      <div
        class="binc-data-tabs__item__value"
      >
        100000000
      </div>
    </div>
  </div>
  <div
    class="binc-data-tabs__group"
  >
    <div
      class="binc-data-tabs__item binc-data-tabs__item--active  binc-data-tabs__item--disabled"
      style="width: 120px;"
    >
      <div
        class="binc-data-tabs__item__header"
      >
        <div
          class="binc-data-tabs__item__title"
        >
          条件C
        </div>
        <div>
          <span
            aria-label="down"
            class="anticon anticon-down ant-dropdown-trigger binc-data-tabs__item__arrow"
            role="img"
            tabindex="-1"
          >
            <svg
              aria-hidden="true"
              data-icon="down"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
              />
            </svg>
          </span>
        </div>
      </div>
      <div
        class="binc-data-tabs__item__value"
      >
        456000
      </div>
    </div>
  </div>
</div>
`;
